clients/python/Makefile

```diff
@@ -16,22 +16,39 @@ clean:
 .PHONY: deploy-latest-mr
 deploy-latest-mr:
 	cd ../../ && IMG_VERSION=${IMG_VERSION} make image/build ARGS=--load && LOCAL=1 ./scripts/deploy_on_kind.sh
-	kubectl port-forward -n kubeflow services/model-registry-service 8080:8080 &
+	kubectl port-forward -n kubeflow services/model-registry-service 8080:8080 & echo $$! >> .port-forwards.pid
 
 .PHONY: deploy-test-minio
 deploy-test-minio:
 	cd ../../ && ./scripts/deploy_minio_on_kind.sh
-	kubectl port-forward -n minio svc/minio 9000:9000 &
+	kubectl port-forward -n minio svc/minio 9000:9000 & echo $$! >> .port-forwards.pid
 
 .PHONY: deploy-local-registry
 deploy-local-registry:
 	cd ../../ && ./scripts/deploy_local_kind_registry.sh
+	kubectl port-forward service/distribution-registry-test-service 5001:5001 & echo $$! >> .port-forwards.pid
 
 .PHONY: test-e2e
 test-e2e: deploy-latest-mr deploy-local-registry deploy-test-minio
+	@echo "Starting test-e2e"
+	-$(MAKE) test-e2e-run; STATUS=$$?
+	$(MAKE) test-e2e-cleanup
+	@exit $$STATUS
+
+.PHONY: test-e2e-run
+test-e2e-run:
+	@echo "Running tests..."
 	@set -a; . ../../scripts/manifests/minio/.env; set +a; \
 	poetry run pytest --e2e -s -rA
-	rm -fr ../../scripts/manifests/minio/.env
+	@rm -f ../../scripts/manifests/minio/.env
+
+.PHONY: test-e2e-cleanup
+test-e2e-cleanup:
+	@echo "Cleaning up port-forward processes..."
+	@if [ -f .port-forwards.pid ]; then \
+		kill $$(cat .port-forwards.pid) || true; \
+		rm -f .port-forwards.pid; \
+	fi
```

Comment:
This is the PID killing step. This cleanup step being separated out also allows us to expand this to clean other things up if needed in the future.